cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/OptimizationVerifier.java

```diff
@@ -93,12 +95,38 @@ static boolean executeGoalsFor(BalancingConstraint constraint,
    * @param verifications      The verifications to make after the optimization.
    * @return Pass / fail status of a test.
    */
-  @SuppressWarnings("unchecked")
   static boolean executeGoalsFor(BalancingConstraint constraint,
                                  ClusterModel clusterModel,
                                  List<String> goalNameByPriority,
                                  Collection<String> excludedTopics,
                                  List<Verification> verifications) throws Exception {
+    return executeGoalsFor(constraint, clusterModel, goalNameByPriority, excludedTopics, verifications, false);
+  }
+
+  /**
+   * Execute given goals in the given cluster enforcing the given constraint. Return pass / fail status of a test.
+   * A test fails if:
+   * 1) Rebalance: During the optimization process, optimization of a goal leads to a worse cluster state (in terms of
+   * the requirements of the same goal) than the cluster state just before starting the optimization.
+   * 2) Self Healing: There are replicas on dead brokers after self healing or soft goal(if any) moves non-immigrant
+   * online replicas.
+   * 3) Adding a new broker causes the replicas to move among old brokers.
```

Comment:
Nit: Looks like the following JavaDoc is repeated in each `#executeGoalsFor`:
```
   * Execute given goals in the given cluster enforcing the given constraint. Return pass / fail status of a test.
   * A test fails if:
   * 1) Rebalance: During the optimization process, optimization of a goal leads to a worse cluster state (in terms of
   * the requirements of the same goal) than the cluster state just before starting the optimization.
   * 2) Self Healing: There are replicas on dead brokers after self healing or soft goal(if any) moves non-immigrant
   * online replicas.
   * 3) Adding a new broker causes the replicas to move among old brokers.
```
For conciseness, should we refer to this JavaDoc from the other two `#executeGoalsFor` for the excerpt above? -- i.e. we can say something like:
```
See {@link #executeGoalsFor(BalancingConstraint, ClusterModel, List, Collection, List, boolean)}.
```